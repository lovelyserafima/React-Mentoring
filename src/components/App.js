import React, { Component } from 'react';
import ErrorBoundary from "./ErrorBoundary";
import MainPage from "./pages/searchpage/MainPage";
import {DetailPageContainer} from "./pages/detailpage/DetailPage";
import {FooterWrapper} from "./App.Styles";
import GlobalStyle from "./Global.Styles";
import PageName from "./header/pagename/PageName";
import { connect } from 'react-redux';

const mapStateToProps = state => ({
    selectedMovie: state.movieReducer.selectedMovie
});

export class App extends Component {

    render() {
        const { selectedMovie } = this.props;
        {console.log(selectedMovie)}
        return (
            <ErrorBoundary>
                <GlobalStyle />
                {selectedMovie === '' ? (
                    <MainPage />
                ) : (
                    <DetailPageContainer movieData={selectedMovie} />
                )}

                <FooterWrapper>
                    <PageName name={'netflixroulette'}/>
                </FooterWrapper>
            </ErrorBoundary>
        );
    }
}

export default connect(mapStateToProps)(App);