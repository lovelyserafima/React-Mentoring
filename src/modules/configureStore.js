import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../modules';

export default (initialState) => {
    return createStore(rootReducer, initialState, applyMiddleware(thunk));
};