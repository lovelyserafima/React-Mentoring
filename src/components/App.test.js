import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';

import { App } from './App';

const mockStore = configureStore([]);

describe('App', () => {
    test('should render correctly', () => {
        const component = shallow(<App />);

        expect(component).toMatchSnapshot();
    });

    it('should render MainPage as default', () => {
        const component = mount(<App />);
        expect(component.find('MainPage')).toHaveLength(1);
        component.unmount();
    });

    it('should render Detail when state.pageType is detail', () => {
        const component = mount(<App />);
        expect(component.find('MainPage')).toHaveLength(1);
        expect(component.find('DetailPage')).toHaveLength(0);
        component.setState({ pageType: 'detail' });
        expect(component.find('MainPage')).toHaveLength(0);
        expect(component.find('DetailPage')).toHaveLength(1);
        component.unmount();
    });
});