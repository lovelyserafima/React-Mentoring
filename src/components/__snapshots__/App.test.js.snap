// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ErrorBoundary>
  <GlobalStyleComponent />
  <Connect(DetailPage) />
  <AppStyles__FooterWrapper>
    <PageName
      name="netflixroulette"
    />
  </AppStyles__FooterWrapper>
</ErrorBoundary>
`;
